draft-ietf-masque-h3-datagram.md

```diff
@@ -303,270 +202,84 @@ Capsule Value:
-  Close Code (i),
-  Close Details (..),
-}
+Capsule-Protocol = sf-boolean
 ~~~
-{: #close-capsule-format title="CLOSE_DATAGRAM_CONTEXT Capsule Format"}
-
-Context ID:
-
-: The context ID to close.
-
-Close Code:
-
-: The close code allows an endpoint to provide additional information as to why
-a datagram context was closed. {{close-codes}} defines a set of codes, the
-circumstances under which an implementation sends them, and how receivers react.
-
-Close Details:
-
-: This is meant for debugging purposes. It consists of a human-readable string
-encoded in UTF-8.
-
-Note that this close is unilateral and bidirectional: the sender of the frame
-unilaterally informs its peer of the closure. Endpoints can use
-CLOSE_DATAGRAM_CONTEXT capsules to close a context that was initially
-registered by either themselves, or by their peer. Endpoints MAY use the
-CLOSE_DATAGRAM_CONTEXT capsule to immediately reject a context that was just
-registered using a REGISTER_DATAGRAM_CONTEXT capsule if they find its Datagram
-Format Type field to be unacceptable.
-
-After an endpoint has either sent or received a CLOSE_DATAGRAM_CONTEXT frame,
-it MUST NOT send any HTTP Datagrams with that Context ID. However, due to
-reordering, an endpoint that receives an HTTP Datagram with a closed Context ID
-MUST NOT treat it as an error, it SHALL instead drop the HTTP Datagram
-silently.
-
-Endpoints MUST NOT close a Context ID that was not previously registered.
-Endpoints MUST NOT close a Context ID that has already been closed. If an
-endpoint receives a CLOSE_DATAGRAM_CONTEXT capsule that violates one or more of
-these requirements, the endpoint MUST abruptly terminate the corresponding
-stream with a stream error of type H3_GENERAL_PROTOCOL_ERROR.
-
-
-#### Close Codes {#close-codes}
-
-Close codes are intended to allow implementations to react differently when they
-receive them - for example, some close codes require the receiver to not open
-another context under certain conditions.
 
-This specification defines the close codes below. Their numeric values are in
-{{iana-close-codes}}. Extensions to this mechanism MAY define new close codes
-and they SHOULD state how receivers react to them.
+Endpoints indicate that the Capsule Protocol is in use on the data stream by
+sending the Capsule-Protocol header field with a value of ?1. A Capsule-Protocol
+header field with a value of ?0 has the same semantics as when the header is not
+present. Intermediaries MAY use this header field to allow processing of HTTP
+Datagrams for unknown HTTP methods or unknown HTTP Upgrade Tokens.
 
-NO_ERROR:
+The Capsule-Protocol header field MUST NOT be sent multiple times on a message.
+The Capsule-Protocol header field MUST NOT be used on HTTP responses with a
+status code different from 2xx (Successful). This specification does not define
+any parameters for the Capsule-Protocol header field value, but future documents
+MAY define parameters. Receivers MUST ignore unknown parameters.
```

Comment:
I am confused by this text.  sf-boolean is literally defined to be either `?0` or `?1`, there are no parameters involved.

Reply:
sf-boolean is a subclass of sf-item which allows parameters
https://datatracker.ietf.org/doc/html/rfc8941#section-3.1.2